osu.Game/OsuGameBase.cs

```diff
@@ -678,12 +678,14 @@ private void onRulesetChanged(ValueChangedEvent<RulesetInfo> r)
         /// <summary>
         /// Allows a maximum of one unhandled exception, per second of execution.
         /// </summary>
-        private bool onExceptionThrown(Exception _)
+        private bool onExceptionThrown(Exception ex)
```

Comment:
Proposing the following:

```diff
diff --git a/osu.Game/OsuGameBase.cs b/osu.Game/OsuGameBase.cs
index 8bda8fb6c2..8b672dde98 100644
--- a/osu.Game/OsuGameBase.cs
+++ b/osu.Game/OsuGameBase.cs
@@ -680,16 +680,17 @@ private void onRulesetChanged(ValueChangedEvent<RulesetInfo> r)
         /// </summary>
         private bool onExceptionThrown(Exception ex)
         {
-            bool continueExecution = Interlocked.Decrement(ref allowableExceptions) >= 0;
-
-            Logger.Log($"Unhandled exception has been {(continueExecution ? $"allowed with {allowableExceptions} more allowable exceptions" : "denied")} .");
-
-            RulesetStore.TryDisableCustomRulesetsCausing(ex);
+            if (Interlocked.Decrement(ref allowableExceptions) <= 0)
+            {
+                Logger.Log($"Too many unhandled exceptions, crashing out.");
+                RulesetStore.TryDisableCustomRulesetsCausing(ex);
+                return false;
+            }
 
+            Logger.Log($"Unhandled exception has been allowed with {allowableExceptions} more allowable exceptions before crash.");
             // restore the stock of allowable exceptions after a short delay.
             Task.Delay(1000).ContinueWith(_ => Interlocked.Increment(ref allowableExceptions));
-
-            return continueExecution;
+            return true;
         }
 
         protected override void Dispose(bool isDisposing)

```

I don't believe we should disable the custom ruleset on the first exception (which is usually "soft" handled aka we won't crash).

I've also refactored the method to make it easier to follow the logical paths.

Reply:
The patch above has a bug (the condition in the `if` is not properly inverted, it should be `< 0`).

Applied in https://github.com/ppy/osu/pull/27698/commits/fb08d6816ba21cc6cb3351694ff587a2100fc86a with that fixed.